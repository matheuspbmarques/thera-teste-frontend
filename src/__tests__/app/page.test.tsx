import '@testing-library/jest-dom'
import { render } from '@testing-library/react'
import Page from '@/app/page';

describe('Page', () => {
    it ('Renders homepage unchanged', () => {
        const { container } = render(<Page />);

        expect(container).toMatchSnapshot();
    })
})