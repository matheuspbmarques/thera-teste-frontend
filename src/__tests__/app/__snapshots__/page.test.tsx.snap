// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Page Renders homepage unchanged 1`] = `
<div>
  <div
    class="p-6"
  >
    <main
      class="max-w-5xl mx-auto"
    >
      <header
        class="flex flex-col gap-4"
      >
        <div
          class="flex justify-between"
        >
          <h1
            class="font-bold text-2xl"
          >
            Meus Produtos
          </h1>
          <button
            class="bg-blue-700 text-slate-100 rounded-full p-2 duration-300 undefined"
            type="button"
          >
            <svg
              fill="none"
              height="24"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
              viewBox="0 0 24 24"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <line
                x1="12"
                x2="12"
                y1="5"
                y2="19"
              />
              <line
                x1="5"
                x2="19"
                y1="12"
                y2="12"
              />
            </svg>
          </button>
        </div>
        <div
          class="flex sm:flex-col gap-2"
        >
          <form
            class="flex min-w-0 bg-white rounded-lg outline outline-gray-300 hover:outline-black has-focus:outline-blue-700 duration-300 flex-1"
          >
            <input
              class="flex-1 py-2 pl-2 outline-none min-w-0"
              name="search"
              placeholder="Procurar..."
              type="search"
              value=""
            />
            <button
              class="p-2 hover:text-blue-700 duration-300"
              type="submit"
            >
              <svg
                fill="none"
                height="24"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
                viewBox="0 0 24 24"
                width="24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <circle
                  cx="11"
                  cy="11"
                  r="8"
                />
                <line
                  x1="21"
                  x2="16.65"
                  y1="21"
                  y2="16.65"
                />
              </svg>
            </button>
          </form>
          <div
            class="flex sm:hidden"
          >
            <button
              class="bg-blue-700 text-slate-100 rounded-full p-2 duration-300 undefined"
              type="button"
            >
              <svg
                fill="none"
                height="24"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
                viewBox="0 0 24 24"
                width="24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <polygon
                  points="22 3 2 3 10 12.46 10 19 14 21 14 12.46 22 3"
                />
              </svg>
            </button>
            <div
              class="fixed left-0 top-0 bg-black/25 w-full h-dvh z-10 px-6"
              style="display: none;"
            >
              <form
                class="bg-white p-4 rounded-2xl m-auto w-full flex flex-col gap-4"
              >
                <header
                  class="flex items-center"
                >
                  <p
                    class="text-xl font-bold flex-1"
                  >
                    Filtrar
                  </p>
                  <button
                    class="bg-blue-700 text-slate-100 rounded-full p-2 duration-300 bg-rose-700 hover:bg-rose-500"
                    type="button"
                  >
                    <svg
                      fill="none"
                      height="24"
                      stroke="currentColor"
                      stroke-linecap="round"
                      stroke-linejoin="round"
                      stroke-width="2"
                      viewBox="0 0 24 24"
                      width="24"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <line
                        x1="18"
                        x2="6"
                        y1="6"
                        y2="18"
                      />
                      <line
                        x1="6"
                        x2="18"
                        y1="6"
                        y2="18"
                      />
                    </svg>
                  </button>
                </header>
                <div
                  class="flex flex-col gap-2"
                >
                  <div
                    class="flex flex-col gap-1 flex-1"
                  >
                    <label>
                      Valor Mínimo
                    </label>
                    <input
                      class="bg-white p-2 outline outline-gray-300 rounded-lg hover:outline-black focus:outline-blue-700 duration-300 w-full"
                      name="minPrice"
                      placeholder="10,00"
                      type="text"
                      value=""
                    />
                  </div>
                  <div
                    class="flex flex-col gap-1 flex-1"
                  >
                    <label>
                      Valor Máximo
                    </label>
                    <input
                      class="bg-white p-2 outline outline-gray-300 rounded-lg hover:outline-black focus:outline-blue-700 duration-300 w-full"
                      name="maxPrice"
                      placeholder="100,00"
                      type="text"
                      value=""
                    />
                  </div>
                  <div
                    class="flex flex-col gap-1 flex-1"
                  >
                    <label>
                      Ordenar preço
                    </label>
                    <div
                      class="bg-white p-2 outline outline-gray-300 rounded-lg flex gap-1 cursor-default relative hover:outline-black duration-300"
                    >
                      <span
                        class="flex-1"
                      >
                        Ordenar de
                      </span>
                      <svg
                        class="text-gray-400 mt-[1px]"
                        fill="none"
                        height="24"
                        stroke="currentColor"
                        stroke-linecap="round"
                        stroke-linejoin="round"
                        stroke-width="2"
                        viewBox="0 0 24 24"
                        width="24"
                        xmlns="http://www.w3.org/2000/svg"
                      >
                        <polyline
                          points="6 9 12 15 18 9"
                        />
                      </svg>
                    </div>
                    <input
                      hidden=""
                      name="priceOrder"
                      type="text"
                      value=""
                    />
                  </div>
                  <button
                    class="flex gap-2 p-2 bg-blue-700 text-slate-100 rounded-lg items-center justify-center hover:bg-blue-500 duration-300 outline-none"
                    type="submit"
                  >
                    <span>
                      Filtrar
                    </span>
                  </button>
                </div>
              </form>
            </div>
          </div>
          <form
            class="flex-1 gap-2 hidden sm:flex"
          >
            <div
              class="flex gap-2 flex-1"
            >
              <div
                class="flex flex-col gap-1 flex-1"
              >
                <input
                  class="bg-white p-2 outline outline-gray-300 rounded-lg hover:outline-black focus:outline-blue-700 duration-300 w-full"
                  name="minPrice"
                  placeholder="Valor Mínimo"
                  type="text"
                  value=""
                />
              </div>
              <div
                class="flex flex-col gap-1 flex-1"
              >
                <input
                  class="bg-white p-2 outline outline-gray-300 rounded-lg hover:outline-black focus:outline-blue-700 duration-300 w-full"
                  name="maxPrice"
                  placeholder="Valor Máximo"
                  type="text"
                  value=""
                />
              </div>
              <div
                class="flex flex-col gap-1 flex-1"
              >
                <div
                  class="bg-white p-2 outline outline-gray-300 rounded-lg flex gap-1 cursor-default relative hover:outline-black duration-300"
                >
                  <span
                    class="flex-1"
                  >
                    Ordenar de
                  </span>
                  <svg
                    class="text-gray-400 mt-[1px]"
                    fill="none"
                    height="24"
                    stroke="currentColor"
                    stroke-linecap="round"
                    stroke-linejoin="round"
                    stroke-width="2"
                    viewBox="0 0 24 24"
                    width="24"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <polyline
                      points="6 9 12 15 18 9"
                    />
                  </svg>
                </div>
                <input
                  hidden=""
                  name="priceOrder"
                  type="text"
                  value=""
                />
              </div>
            </div>
            <button
              class="flex gap-2 p-2 bg-blue-700 text-slate-100 rounded-lg items-center undefined hover:bg-blue-500 duration-300 outline-none"
              type="submit"
            >
              <span>
                Filtrar
              </span>
            </button>
          </form>
        </div>
      </header>
      <div>
        <img
          alt="Sem Produtos"
          class="mx-auto mt-6"
          data-nimg="1"
          decoding="async"
          height="256"
          loading="lazy"
          src="/_next/image?url=%2Fimg.jpg&w=640&q=75"
          srcset="/_next/image?url=%2Fimg.jpg&w=256&q=75 1x, /_next/image?url=%2Fimg.jpg&w=640&q=75 2x"
          style="color: transparent;"
          width="256"
        />
        <p
          class="text-center max-w-80 mx-auto"
        >
          Ops! Você ainda não tem nenhum produto cadastrado
        </p>
      </div>
      <div
        class="mt-6 flex justify-center gap-2"
      />
    </main>
  </div>
</div>
`;
